### PR TITLE
Get all existing tests passing for sum and add two arg tests.

### PR DESCRIPTION
**Pull request by martica**
Fills out the rest of the functionality and error cases of sum().

## PR Checklist:

- [x] All new features have been tested
- [ ] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

File: tests/builtins/test_sum.py [hunk 1]
```diff
@@ -46,6 +51,38 @@ class BuiltinSumFunctionTests(BuiltinFunctionTestCase, TranspileTestCase):
...
-    not_implemented = [
-        'test_range',  # This has been implemented, but fails upstream on isinstance.
+
+class BuiltinSumTwoArgFunctionTests(BuiltinTwoargFunctionTestCase, TranspileTestCase):
+    function = "sum"
+
+    # these are implemented, but the exact exception thrown depends on the order
+    # that they are iterated on being the exact same in both CPython and batavia,
+    # which is not guaranteed. (if an unsupported string follows an int, the error
+    # will be different than if it followed a float)
```
Comment (freakboy3742) [inline tests/builtins/test_sum.py]: This is a familiar problem; historically, it's what we've used the "substitutions" feature of the test tools to fix. If you enumerate all possible orderings, and then use substitution to normalize to a consistent ordering, these tests can be made to pass.

File: tests/builtins/test_sum.py [hunk 2]
```diff
@@ -32,20 +31,36 @@ def test_sum_mix_floats_and_ints(self):
...
+        """)
+
 
 class BuiltinSumFunctionTests(BuiltinFunctionTestCase, TranspileTestCase):
     function = "sum"
+    substitutions = SAMPLE_SUBSTITUTIONS.copy()
+    substitutions.update({
+        'unsupported operand type(s) for +: other type and one of float, str or int': [
+            re.compile(
+                r"<class 'TypeError'> : unsupported operand type\(s\) for \+: '(float|str|int)' and '(float|str|int)'"),
```
Comment (brutusthebee) [inline tests/builtins/test_sum.py]: At column 120: (E501) line too long (120 > 119 characters)

File: tests/builtins/test_sum.py (hunk 1 above)
Comment (martica) [reply tests/builtins/test_sum.py]: Good call. Updated.

**Comment by martica** [comment]
@freakboy3742 I've fixed the flakey tests. Can you take another look?

**Pull request merged**